src/hnsw/build.c

```diff
@@ -482,25 +492,50 @@ static void BuildIndex(
         assert(error == NULL);
     }
 
-    char *result_buf = NULL;
-    usearch_save(buildstate->usearch_index, NULL, &result_buf, &error);
-    assert(error == NULL && result_buf != NULL);
-
-    size_t num_added_vectors = usearch_size(buildstate->usearch_index, &error);
+    metadata = usearch_metadata(buildstate->usearch_index, &error);
     assert(error == NULL);
 
+    if(buildstate->index_file_path == NULL) {
+        // size of static name + max digits of uint32 (Oid) + 1 for nullbyte
+        index_file_path = palloc0(sizeof("/tmp/ldb-index-.bin") + 10 + 1);
```

Comment:
* Factor out the calculation and the format string constant to a local variable
* use the constant format string in the sprintf call, so if we change the file path, we will be sure both allocation and formatting will be updated
* use snprintf or other sized string formatter to make sure we do not silently buffer-overflow
* make sure the size parameter passed to snprintf is a compile time constant (it is in this case as it is sizeof() + constant)